src/main/java/com/matera/cadastrocentral/client/ClientService.java

```diff
@@ -0,0 +1,109 @@
+package com.matera.cadastrocentral.client;
+
+import com.matera.cadastrocentral.identitydocument.IdentityDocumentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ClientService {
+
+    private final ClientRepository clientRepository;
+    private final IdentityDocumentRepository identityDocumentRepository;
+
+    @Autowired
+    public ClientService(final ClientRepository clientRepository,
+                         final IdentityDocumentRepository identityDocumentRepository) {
+        this.clientRepository = clientRepository;
+        this.identityDocumentRepository = identityDocumentRepository;
+    }
+
+    /* API requests */
+
+    // 1. Get all clients from the database.
+    public List<Client> getAllClients() {
+        return clientRepository.findAll();
+    }
+
+    // 2. Get a specific client by id.
+    public Optional<Client> getClientById(UUID clientId) {
+        Optional<Client> client = clientRepository.findById(clientId);
+        if(client.isPresent()){
+            return client;
+        } else {
+            throw new ResponseStatusException(
+                    HttpStatus.NOT_FOUND,
+                    "Client ID does not exist in the database! Try a valid one."
+            );
+        }
+    }
+
+    // 3. Insert a client into the database.
+    public Client insertClient(final ClientDTO clientDTO) {
+        // Marital Status should be specified in the body
+        // since it has a not nullable constraint
+        if(clientDTO.getMaritalStatusEntity() != null){
+            // Identity Document List should be specified in the body
+            // since it has a not nullable constraint, and it should
+            // contains at least one identity document
+            if(clientDTO.getIdentityDocumentEntityList() != null &&
+                    !clientDTO.getIdentityDocumentEntityList().isEmpty()) {
+                Client client = clientRepository.save(new Client(clientDTO));
```

Comment:
Here you should be able to save a client even if there is no documents on payload, move `...clientRepository.save...` out of the `if(clientDTO.getIdentityDocumentEntityList() != null...`